strongswan/collector.go

```diff
@@ -475,3 +512,67 @@ func viciStateToInt(v string) connectionStatus {
 		return unknown
 	}
 }
+
+func (c *Collector) collectCrtMetrics(crt *Crt, ch chan<- prometheus.Metric) {
+	now := time.Now()
+	valid := 0
+	if now.After(crt.NotBefore) && now.Before(crt.NotAfter) {
+		valid = 1
+	}
+
+	ch <- prometheus.MustNewConstMetric(
+		c.crtValid,
+		prometheus.GaugeValue,
+		float64(valid),
+		crt.Subject, crt.NotBefore.Format(time.RFC3339), crt.NotAfter.Format(time.RFC3339),
+	)
+}
+
+func (c *Collector) listCrts() ([]Crt, error) {
+	s, err := c.viciClientFn()
+	if err != nil {
+		return nil, err
+	}
+	defer s.Close()
+
+	msgs, err := s.StreamedCommandRequest("list-certs", "list-cert", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	res := []Crt{}
+	for _, m := range msgs {
+		if m.Err() != nil {
+			return nil, err
+		}
+
+		val := m.Get("type")
+		cert_type, ok := val.(string)
+		if !ok {
+			continue
+		}
+
+		if cert_type != "X509" {
+			continue
+		}
+
+		val = m.Get("data")
+		data, ok := val.(string)
+		if !ok {
+			return nil, err
+		}
+
+		cert, err := x509.ParseCertificate([]byte(data))
+		if err != nil {
+			return nil, err
+		}
+
+		res = append(res, Crt{
+			Subject:   cert.Subject.String(),
+			NotBefore: cert.NotBefore,
+			NotAfter:  cert.NotAfter,
+		})
```

Comment:
I would rather parse all the attributes in a raw form. Then you can filter and process `X509` type later in  `collectX509CrtMetrics`.